wdl/downsampling_pipeline_part2.wdl

```diff
@@ -3,7 +3,7 @@ version 1.0
 workflow downSampling_02 {
```

Comment:
Comments for 02 WDL:
- countAndRandomSample: 
  - Does this preserve read pairs? It looks like fastq-sample has this capability if you provide 2 fastq files to it at once, but it looks like you’re doing each end of the pair separately - have you checked that this method preserves pairs?
  - I'm a bit confused about Float start_depth, Float final_depth - are depths floats? Why do you need the start_depth? How do you have the start_depth at this point in the pipeline? Maybe some comments would help here
  - I'm also not really sure what calculations are happening (in this task and in the scripts it calls) so some comments would also help to explain that
- realign: line 281 - `nproc` is not defined (side note: if you make this an input, make sure to make it have a default value, just like the resource attributes) 
- addRG: 
  - [thinking aloud] Is there a way to avoid having to input the original cram, so that it doesn't have to be copied into a new task (both the copy operation and the additional disk space to store it cost money)? I see that it's used to add the read group back, but where does the read group disappear? Is it just not in the fastq (and is there a way to preserve it)? Or maybe you could extract the RG info in downsampling_01 in a task where you already have to have the bam or cram and then output it in a file or as a string, then use -R in bwa mem to add it? 
  - Also, the input is named cram, but does it work if it’s a bam not a cram from the start? Would be good to make sure that it works for both bam & cram, and update the name or add a comment if it can be either
  - There's some repeated commands, so maybe just do one `samtools view | grep` to get the RG and then extract the relevant field from that string? I guess if that part is fast then it doesn’t matter, but should make sure it's not running inefficiently
- Some of the task names aren’t very descriptive - addRG also sorts, markDuplicates also contains bam to cram, etc. If these tasks are long you might as well break them up into distinct tasks by actions (good from a code style standpoint to be more modular, but also from a practical/cost standpoint to use more preemptible VMs). But if they are still pretty quick and run easily on preemptible VMs, it's fine to not break them up (though modularity is still something to keep in mind as a goal), but it would be good to have more descriptive names at least.
- I’m not sure I understand the purpose of all the tasks (markDuplicates, collectCoverage, collectCountsCram) so I just want to be transparent that it’s harder for me to give feedback on the structure of the pipeline 
	- But yeah I am a little confused by the collect coverage/counts at the end, in part because I thought that gCNV did that as part of their pipeline (could be totally mistaken), so will it be repeated compute? Was it just to verify that the downsampling had occurred as expected? Should it be a separate QC-type workflow that doesn’t need to be run every time?
- CollectCountsCram: did you not find that it needed any memory overhead? (to follow up on a previous slack discussion we had)
- Are all of these workflow outputs needed? See my 01 comments on intermediate vs. workflow outputs

Reply:
countAndRandomSample:
- It does preserve read pairs with the same seed, but you are right, they do have read pair functionality so I will modify this to just running fastq-sample once not twice. 
- Float start_depth and and Float final_depth are only floats so the awk line works when I calculate for half of the final and initial read depth so I can sample each fq file for 1/2 the number of reads

addRG:

- Those are pretty short tasks but I will change the task names to be more descriptive
- collectCoverage is used to double check that the downsampled file has the final read depth that we want for each run.
- markDuplicates makes it so that it will mirror closer to what is coming off sequencer
- I believe the output of collectCountsCram is the file that will be used for further downstream analyses, which I am not too sure what it is..
- CollectCountsCram: it runs pretty quick on Terra, ~20 minutes, I will keep you posted about whether it does once cromwell runs
- I'll clean up and only keep the necessary workflow outputs

Reply:
addRG:

- renamed: addReadGroupAndSort, now reads in only read group text file extracted from part 1 of workflow